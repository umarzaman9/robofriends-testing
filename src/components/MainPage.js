import React, { Component } from 'react';



import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundry from '../components/ErrorBoundry';
import CounterButton from '../components/CounterButton'
import './MainPage.css';


class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

filterRobots = () => {
    return this.props.robots.filter(robot => {
      return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
    })
}  
  render() {
    const { robots, onSearchChange, isPending } = this.props;
  
    return (
      <div className='tc'>
        <h1 className='f1'>RoboFriends</h1>
        <SearchBox searchChange={onSearchChange}/>
        <CounterButton />
        <Scroll>
          { isPending ? <h1>Loading</h1> :
            <ErrorBoundry>
              <CardList robots={this.filterRobots} />
            </ErrorBoundry>
          }
        </Scroll>
      </div>
    );
  }
}
export default MainPage;
