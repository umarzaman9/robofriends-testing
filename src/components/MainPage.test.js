import React from 'react';
import {shallow} from 'enzyme';
import MainPage from './MainPage';

let wrapper; 
let wrapper2;
beforeEach(()=> {
	const mockProps = {
		onRequestRobots: jest.fn(),
		robots: [],
		searchField: '',
		isPending: false
	}
	wrapper = shallow(<MainPage {...mockProps} />)
})

it('render MainPage without crashing', ()=> {
	expect(wrapper).toMatchSnapshot();
})

it('filter the robot now', () => {
	const mockProps2 = {
		onRequestRobots: jest.fn(),
		robots: [{
			id: 3,
			name: 'John',
			email: 'john@gmail.com'
		}],
		searchField: 'john',
		isPending: false
	}
	wrapper2 = shallow(<MainPage {...mockProps2} />)
	expect(wrapper2.instance().filterRobots()).toEqual([{
			id: 3,
			name: 'John',
			email: 'john@gmail.com'
		}])
})