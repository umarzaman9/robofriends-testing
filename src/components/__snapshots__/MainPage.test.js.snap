// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MainPage without crashing 1`] = `ShallowWrapper {}`;
